src/wp-includes/blocks.php

```diff
@@ -252,12 +261,26 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		);
 	}
 
-	return register_block_type(
-		$metadata['name'],
+	/**
+	 * Filters the settings determined from the block type metadata.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param array $settings Array of determined settings for registering a block type.
+	 * @param array $metadata Metadata provided for registering a block type.
+	 */
+	$settings = apply_filters(
+		'post_register_block_type_from_metadata',
```

Comment:
Similarly, this is misleading, as one could think this refers to the "post" post type. WordPress does not use `pre_` and `post_` companion filters like this anywhere either.

Perhaps just `block_type_metadata_settings`?